### PR TITLE
fix: undo and redo broken in webviews

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/22744.

When propagating the undo and redo events from the menu, Chromium was not
properly sending the events to the focused webcontents. This fixes that by ensuring that the events are sent to the proper webcontents.

cc @zcbenz @MarshallOfSound @nornagon 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where undo and redo shortcuts did not work in webviews.

Co-authored-by: Samuel Attard <sattard@slack-corp.com>

### CODE REVIEW

**Comment by deepak1556** [comment]
Is this only an issue for macOS ?

**Comment by codebytere** [comment]
@deepak1556 yes i believe so, see [this comment](https://github.com/electron/electron/issues/22744#issuecomment-601492527).

**Comment by deepak1556** [comment]
Thanks, can you mention the platform affected in the release notes. The notes reflects webview in general is affected, so can assume all platforms.

**Comment by deepak1556** [comment]
Just tested the following sample on web with OOPIF and in electron with webview. The problem is not seen with OOPIF, I don't think the patch completely identifies the issue , we seem to workaround it.

```html
// main.html
<iframe src="http://localhost:8000/test.html"></iframe>
```

```html
//embed.html
<input type="text"/>
```

**Comment by codebytere** [comment]
@deepak1556 would you prefer we not merge this solution and try to find another one? Or shall we merge this and see if other issues come up later and we can regroup then?

**Comment by deepak1556** [comment]
I would suggest to identify the reason why it works for plain OOPIF and not for our webview implemented on top of OOPIF. I am not comfortable merging the solution in its current state, a more closer understanding of the problem would help.

**Comment by MarshallOfSound** [comment]
@deepak1556 I believe this patch is correct, OOPIF are not separate webContents.  WebViews are implemented as separate webContents that are attached via "inner" webContents.

The documentation for `GetFocusedWebContents` says this is explicitly what we want when taking into account guest views and inner web contents.

**Comment by deepak1556** [comment]
@MarshallOfSound thanks, that explains it.

I was able to repro the issue with chrome's webview https://developer.chrome.com/apps/tags/webview which is based on `//compoent/guest_view` that is similar design to our webview. Can you please open an upstream issue and link it in the patch. Thanks!

File: patches/chromium/fix_undo_redo_broken_in_webviews.patch [hunk 1]
```diff
@@ -5,7 +5,8 @@ Subject: fix: undo redo broken in webviews
 
 When propagating the undo and redo events from the menu, Chromium was not
 properly sending the events to the focused webcontents on macOS. This fixes
-that error, and will be upstreamed into Chromium.
+that error. A crbug has been opened at https://bugs.chromium.org/p/chromium/issues/detail?id=1067284
```
Comment (deepak1556) [inline patches/chromium/fix_undo_redo_broken_in_webviews.patch:8]: Thank you! 🙇

**Comment by akashnimare** [comment]
@sofianguy is this being added in the v9?

**Comment by ckerr** [comment]
It's been awhile since the question was asked so @akashnimare you've probably already found the answer, but I'll add this comment in case anyone else comes across this PR in the future:

https://github.com/electron/electron/pull/23370, which did land in 9-x-y, superceded this PR

**Pull request merged**